draft-ietf-core-corr-clar.md

```diff
@@ -241,6 +249,142 @@ Further discussion of a more generalized response concept can be found in
+and Location-Query) of -coap}}.)
+
+Similarly, {{Section 6.1 (coap URI Scheme) of -coap}} says:
+
+{:quote}
+>   The query serves to further parameterize the resource.  \[...]
+
+This could be read to say that the *same* resource can be accessed
+with different parameters in the Uri-Query options.
+It is likely that these passages were meant in the sense of
+"parameterize the resource name", i.e., changing the parameters does
+indeed lead to a different resource.
+
+So this could be clarified by applying this change in these three
+places:
+
+INCORRECT:
+: further parameterize the resource
+
+CORRECTED:
+: further parameterize the resource name
+
+However, the view that the query component supplies parameters to a
+request on a resource that exists independent of these parameters is
+widely held by implementers in the "big web" (HTTP) world, even if
+{{-http}} strictly follows {{-uri}}.
+
+This view seems to be fueled by the way that an application may use
+({{Section 17.9 (Disclosure of Sensitive Information in URIs) of -http}})...
+
+{:quote}
+> client-side mechanisms to construct a target URI out of
+  user-provided information, such as the query fields of a form using
+  GET
+
+This view also seems to be suggested to some by the way query
+parameters are used in specifications such as {{-cond}}; implementations
+of CoAP servers also often provide primitives to set up a single
+"resource handler" that bundles requests to a specific path and receives the
+query parameters as additional request parameters {{COAP-RESOURCE}}.
+
+{{-lawn}} establishes guidelines with respect to "URI Design and
+Ownership".
+{{Section 2.4 of RFC8820@-lawn}} specifically discusses the query
+component of the URI.
+On one hand, it says:
+
+{:quote}
+> Applications can specify the syntax of queries for the resources under their control.
+
+On the other hand, it warns:
+
+{:quote}
+> Extensions MUST NOT constrain the format or semantics of queries, to avoid collisions and erroneous client assumptions. For example, an Extension that indicates that all query parameters with the name "sig" indicate a cryptographic signature would collide with potentially preexisting query parameters on sites and lead clients to assume that any matching query parameter is a signature.
+
+It also acknowledges:
+
+{:quote}
+> Per the "Form submission" section of [HTML5], HTML constrains the syntax of query strings used in form submission. New form languages are encouraged to allow creation of a broader variety of URIs (e.g., by allowing the form to create new path components, and so forth).
+
+{{-cond}} is a specification that defines a set of query parameters.
+It can be adopted by an "Application" in the sense of {{-lawn}}.
+The current discussion goes in the direction of providing an interface
+type (for use in {{-link-format}} `if=`) that a server can declare in resource discovery to indicate
+to a client that the conditional query parameters interface is available.
```

Comment:
Just as a remark: "Application" is really fitting, although that only works in German, where it invokes the image of an [Appliqué](https://en.wikipedia.org/wiki/Appliqu%C3%A9) that the designer chooses to attach to the resource.